lms/services/user.py

```diff
@@ -34,11 +34,33 @@ def store_lti_user(self, lti_user: LTIUser):
             existing_user.roles = new_user.roles
 
         else:
-            self.db_session.add(new_user)
+            self._db.add(new_user)
+
+    @lru_cache
+    def get(self, application_instance, user_id: str) -> User:
+        """
+        Get a User that belongs to `application_instance` an has user_id=user_id.
+
+        :param application_instance: The ApplicationInstance the user belongs to
+        :param user_id: Unique identifier of the user
+        :raises UserNotFound: if the User is not present in the DB
+        """
+        existing_user = (
+            self._db.query(User)
+            .filter_by(
+                application_instance=application_instance,
+                user_id=user_id,
+            )
+            .one_or_none()
+        )
+        if not existing_user:
+            raise UserNotFound()
+
+        return existing_user
```

Comment:
I think you might be able to use [`one()`](https://docs.sqlalchemy.org/en/14/core/connections.html?highlight=one_or_none#sqlalchemy.engine.CursorResult.one) here instead for slightly simpler code:

```python
try:
    return self._db.query(User).filter_by(...).one()
except NoResultFound as err:
    raise UserNotFound() from err
```